### ISSUE TITLE
More Flexible RequestPathRule

### ISSUE THREAD

**Issue by hguiney** [tuupola/slim-jwt-auth#4]
Right now I can authenticate for a top-level route and passthrough sub-routes. From the documentation:
```
new RequestPathRule([
    "path" => "/api",
    "passthrough" => ["/api/token", "/api/ping"]
]),
```
But what I want is the reverse, to authenticate sub-routes, and passthrough a top-level route. I realize this is an unrealistic example, but just for comparison's sake:

```
new RequestPathRule([
    "path" => ["/api/token", "/api/ping"]
    "passthrough" => ["/api"]
]),
```
This would allow me to do things like `POST /users/` unauthenticated, for sign-ups, but then require requests to `GET /users/me` to be authenticated.

**Comment by hguiney** [comment]
Also, when a route is set up as a passthrough, the callback function is not executed, so the `$app->jwt` variable is not set, even if the client is sending an `Authentication` header. However, I have a temporary workaround:
```
if ( !isset( $app->jwt ) ) {
    $headers = $app->request->headers;
    $authorization = $app->request->headers->get( 'Authorization' );

    if ( $authorization ) {
      $jwtEncoded = substr( $authorization, strlen( 'Bearer ' ) );
      $secret = base64_decode( JWT_SECRET );
      $jwt = JWT::decode( $jwtEncoded, $secret, JWT_ALGOS );

      $app->jwt = $jwt;
    } else {
      // error
    }
}
```

**Comment by tuupola** [comment]
I probably change `path` to accept an array later. But currently you should be able to add two request path rules. 

```php
$app->add(new JwtAuthentication([
    "secret" => getenv("JWT_SECRET"),
    "rules" => [
        new RequestPathRule([
            "path" => "/api/token"
        ]),
        new RequestPathRule([
            "path" => "/api/ping"
        ])
    ]
]));
```

Now everything under `/api/token` and `/api/ping` are authenticated but for example `/api/foo` is not. Or just write your own rules. They are simple callables.

**Comment by tuupola** [comment]
Check [development branch](https://github.com/tuupola/slim-jwt-auth/tree/development). It now allows you to do stuff like.

```php
$auth = new JwtAuthentication([
    "secret" => getenv("JWT_SECRET"),
    "path" => ["/api/foo", "/api/bar"]
]);
```

```php
$app->add(new JwtAuthentication([
    "secret" => getenv("JWT_SECRET"),
    "rules" => [
        new RequestPathRule([
            "path" => ["/api/foo", "/api/bar"]
        ])
    ]
]));
```

Will do a release after I have run new code myself for a while.

**Comment by hguiney** [comment]
Terrific, thank you!

**Comment by tuupola** [comment]
Released in [0.4.0](https://github.com/tuupola/slim-jwt-auth/releases/tag/0.4.0).
